### PR TITLE
refactor: use native Svelte stores instead of third-party library

### LINKED ISSUE #379: Use svelte stores instead of macfja/svelte-persistent-store
### Is your feature request related to a problem? Please describe.

Why use a third-party library when we can use svelte's own? 

### Describe the solution you'd like

Use svelte stores and change js-files to ts

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Pankai222**
# Description

Refactors macfja/svelte-persistent-stores into svelte stores and uses typescript and places each store into either preferences or cookies and basically you are fucking stupid

Fixes #379

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: frontend/src/lib/components/theme_selector.svelte [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <script lang="ts">
   import { themeChange } from "theme-change"
   import { onMount } from "svelte"
-  import { chosenTheme } from "../stores/theme.js"
+  import { chosenTheme } from "../stores/preferences"
```
Comment (Primdahl26) [inline frontend/src/lib/components/theme_selector.svelte:4]: Can we use `/$lib`?

File: frontend/src/lib/components/country_selector.svelte [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <script lang="ts">
   import { COUNTRIES } from "../variables.js"
-  import { currentCountry } from "../stores/country.js"
+  import { currentCountry } from "../stores/preferences"
```
Comment (Primdahl26) [inline frontend/src/lib/components/country_selector.svelte:3]: Can we use `/$lib`?

File: frontend/src/lib/components/country_locator.svelte [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <script lang="ts">
   import { onMount } from "svelte"
-  import { currentCountry, confirmedCountry } from "../stores/country"
+  import { currentCountry, confirmedCountry } from "../stores/preferences"
```
Comment (Primdahl26) [inline frontend/src/lib/components/country_locator.svelte:3]: Can we use `/$lib`?

File: frontend/src/lib/components/cookie_disclaimer.svelte [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <script lang="ts">
   import { fade, fly } from "svelte/transition"
-  import { cookieDisclaimer } from "../stores/cookie_disclaimer.js"
+  import { cookieDisclaimer } from "../stores/cookies.js"
```
Comment (Primdahl26) [inline frontend/src/lib/components/cookie_disclaimer.svelte:3]: Can we use `/$lib`?

File: frontend/src/lib/components/no_results.svelte [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <script lang="ts">
   import { fade } from "svelte/transition"
   import type { Genre } from "../types"
-  import { filters } from "../stores/filters.js"
+  import { filters } from "../stores/preferences"
```
[the remark this answers is not in the document]
Comment (Pankai222) [reply frontend/src/lib/components/no_results.svelte]: Done

File: frontend/src/lib/components/theme_selector.svelte (hunk 1 above)
Comment (Pankai222) [reply frontend/src/lib/components/theme_selector.svelte]: Done

File: frontend/src/lib/components/country_selector.svelte (hunk 2 above)
Comment (Pankai222) [reply frontend/src/lib/components/country_selector.svelte]: Done

File: frontend/src/lib/components/cookie_disclaimer.svelte (hunk 4 above)
Comment (Pankai222) [reply frontend/src/lib/components/cookie_disclaimer.svelte]: Done

**Pull request closed without merging**